src/foam/u2/view/TreeView.js

```diff
@@ -251,21 +251,27 @@ foam.CLASS({
 
   methods: [
     function initE() {
-      var M   = this.ExpressionsSingleton.create();
-      var of  = this.lookup(this.relationship.sourceModel);
+      var M = this.ExpressionsSingleton.create();
+      var of = this.lookup(this.relationship.sourceModel);
       var dao = this.data$proxy.where(
         M.NOT(M.HAS(of.getAxiomByName(this.relationship.inverseName))));
 
-      var self = this;
-      this.addClass(this.myClass()).
-        select(dao, function(obj) {
-          return self.TreeViewRow.create({
-            data: obj,
-            relationship: self.relationship,
-            expanded: self.startExpanded,
-            formatter: self.formatter
-          }, this);
-        });
+      var element = this.addClass(this.myClass());
+      dao.select().then((sink) => {
```

Comment:
Doing dao.select().then is dramatically different from doing this.select().

In a U2 Element, this.select(dao, function(){}) hooks up dynamically so that it re-renders when the dao changes.

I think what you could do instead and would be a much smaller change would be

```javascript
var first = true;
this.select(dao, function(obj) {
  if ( first && ! self.selection ) self.selection = obj;
  first = false;
  ...
```

That's a much smaller diff and therefore less risk.